lib/commit.js

```diff
@@ -14,6 +14,30 @@ var _amend = Commit.prototype.amend;
  */
 Commit.lookup = LookupWrapper(Commit);
 
+/**
+ * @async
+ * @param {Number} n
+ * @return {Commit}
+ */
+Commit.prototype.parent = (function(parent) {
+  if (typeof parent !== "function") {
+    throw new Error("parent is not a function!");
+  }
+
+  return function(n, callback) {
+    var repo = this.repo;
+    return parent.call(this, n).then(p => {
+      p.repo = repo;
+
+      if (typeof callback === "function") {
+        callback(null, p);
+      }
+
+      return p;
+    }, callback);
+  };
+})(Commit.prototype.parent);
```

Comment:
This is a big change in style from how most of NodeGit is written. I believe we have a pattern for these overrides where we will define:
```javascript
var _parent = Commit.prototype.parent;
```

at the top of the file under _amend.

Then we will reference that later in the file. It's not my favorite scheme, but @maxkorp spent a decent amount of time building a pattern for it that I'm not comfortable tossing out just yet.